apps/ui/src/helpers/stamp.ts

```diff
@@ -1,22 +1,41 @@
 import { formatAddress } from './utils';
 
+const resolvedAddresses = new Map<string, string | null>();
+
 export async function getNames(addresses: string[]): Promise<Record<string, string>> {
   try {
     const inputMapping = Object.fromEntries(
       addresses.map(address => [address, formatAddress(address)])
     );
+    const resolvedAddressesKeys = Array.from(resolvedAddresses.keys());
+    const unresolvedAddresses = Object.values(inputMapping).filter(
+      address => !resolvedAddressesKeys.includes(address)
+    );
+    let data: string[] = [];
+
+    if (unresolvedAddresses.length > 0) {
+      const res = await fetch('https://stamp.fyi', {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json'
+        },
+        body: JSON.stringify({
+          method: 'lookup_addresses',
+          params: unresolvedAddresses
+        })
+      });
+      data = (await res.json()).result;
 
-    const res = await fetch('https://stamp.fyi', {
-      method: 'POST',
-      headers: {
-        'Content-Type': 'application/json'
-      },
-      body: JSON.stringify({ method: 'lookup_addresses', params: Object.values(inputMapping) })
-    });
-    const data = (await res.json()).result;
+      Object.values(inputMapping).forEach((formatted: string) => {
+        resolvedAddresses.set(formatted, data[formatted]);
+      });
+    }
 
     const entries: any = Object.entries(inputMapping)
-      .map(([address, formatted]) => [address, data[formatted] || null])
+      .map(([address, formatted]) => [
+        address,
+        data[formatted] || resolvedAddresses.get(formatted) || null
```

Comment:
```suggestion
        resolvedAddresses.get(formatted) || null
```

In this branch it would already be in resovledAddresses if it was in data.